extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php

```diff
@@ -4,40 +4,54 @@
  * @author Inez Korczyński <korczynski@gmail.com>
  * @author Piotr Bablok <piotr.bablok@gmail.com>
  */
-
 class AssetsManagerSassBuilder extends AssetsManagerBaseBuilder {
 	const CACHE_VERSION = 2;
 
-	public function __construct(WebRequest $request) {
+	/**
+	 * @var bool $exists Whether the file we're trying to load exists
+	 */
+	protected $exists = true;
+
+	public function __construct( WebRequest $request ) {
 		global $IP;
-		parent::__construct($request);
+		parent::__construct( $request );
 
 		// TODO: the background image shouldn't be passed as the url - we should pass a File reference and derive ourselves
-		if (isset($this->mParams['background-image']) && VignetteRequest::isVignetteUrl($this->mParams['background-image']) && isset($this->mParams['path-prefix'])) {
-			$connector = strpos($this->mParams['background-image'], '?') === false ? '?' : '&';
+		if ( isset( $this->mParams['background-image'] ) && VignetteRequest::isVignetteUrl( $this->mParams['background-image'] ) && isset( $this->mParams['path-prefix'] ) ) {
+			$connector = strpos( $this->mParams['background-image'], '?' ) === false ? '?' : '&';
 			$this->mParams['background-image'] .= "{$connector}path-prefix={$this->mParams['path-prefix']}";
 		}
 
-		if (strpos($this->mOid, '..') !== false) {
-			throw new Exception('File path must not contain \'..\'.');
+		if ( strpos( $this->mOid, '..' ) !== false ) {
+			throw new Exception( 'File path must not contain \'..\'.' );
 		}
 
-		if (!endsWith($this->mOid, '.scss', false)) {
-			throw new Exception('Requested file must be .scss.');
+		if ( !endsWith( $this->mOid, '.scss', false ) ) {
+			throw new Exception( 'Requested file must be .scss.' );
 		}
 
 		//remove slashes at the beginning of the string, we need a pure relative path to open the file
 		$this->mOid = preg_replace( '/^[\/]+/', '', $this->mOid );
 
 		if ( !file_exists( "{$IP}/{$this->mOid}" ) ) {
-			throw new Exception("File {$this->mOid} does not exist!");
+			// SUS-399: allow SCSS processing to continue, log the error
+			$this->exists = false;
+			Wikia\Logger\WikiaLogger::instance()->info( 'Nonexistent SCSS file requested', [
+				'oid' => $this->mOid,
+			] );
 		}
 		$this->mContentType = AssetsManager::TYPE_CSS;
 	}
 
 	public function getContent( $processingTimeStart = null ) {
 		global $IP, $wgEnableSASSSourceMaps;
-		wfProfileIn(__METHOD__);
+		wfProfileIn( __METHOD__ );
+
+		// SUS-399: Abort early if we received a request for a file that no longer exists
+		if ( !$this->exists ) {
+			wfProfileOut( __METHOD__ );
+			return '';
+		}
```

Comment:
yes - if this is a group request for multiple sasses, this means the rest will load correctly and we'll just load nothing for the deleted file :)